server/src/main/java/io/crate/user/metadata/RolesMetadata.java

```diff
@@ -67,6 +67,7 @@ public RolesMetadata addOldUsersMetadata(@Nullable UsersMetadata usersMetadata)
         if (usersMetadata == null) {
             return this;
         }
+        roles.clear();
```

Comment:
Can this method be a ctor or factory method instead? then roles must not be cleared... Looks like it's somehow used in this way already somehow by `new RolesMetadata().addOld...`